mlflow/tensorflow.py

```diff
@@ -705,6 +705,36 @@ def fit(self, *args, **kwargs):
         if _AUTO_END_RUN:
             mlflow.end_run()
         _AUTO_END_RUN = False
+
+        return result
+
+    @gorilla.patch(tensorflow.keras.Model)
+    def fit_generator(self, *args, **kwargs):
+        global _AUTO_END_RUN
+        if not mlflow.active_run():
+            mlflow.start_run()
```

Comment:
Actually I realized - `mlflow.start_run` can make an API request, so we should wrap it in a `try_mlflow_log` it along with `mlflow.end_run`, and only attempt end the run if starting the run succeeds. We should also do the same for Keras autologging in a follow-up, sorry I missed this the first time around!

Reply:
Hm thinking about it, there are a few things that could happen:

1) `start_run()` fails due to server unavailability but at least some successive logging calls succeed (could happen e.g. if the server is being updated during the `start_run`)
2) `start_run()` & all successive logging calls fail so autologging never starts a run, but `end_run` succeeds (pretty unlikely given that we typically make several calls between `start_run` and `end_run`)
3) `start_run()` succeeds, but `end_run` fails so we never terminate the run (could happen if the server becomes unavailable by the time the run finishes)

I think 1) & 3) could happen (they're hopefully unlikely) but 2) will almost never happen, so maybe we don't need to do:
>  only attempt to end the run (only set _AUTO_END_RUN = True) if starting the run succeeds

and can simply just wrap the `start_run` and `end_run` in a `try_mlflow_log`